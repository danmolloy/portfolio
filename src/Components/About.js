export const About = () => {
  return (

    <div id="about" className="text-gray-400 h-screen flex flex-col text-center justify-center blue-bg">
      <h2>About</h2>
      <p>Originally from Australia, I moved to London in 2014.</p>
      <p>I have spent the past 12 years working as a freelance double bassist.</p>
      <p>I have performed in some of the world's leading ensembles, such as the London Symphony Orchestra.</p>
      <p>I'm on a variety of soundtracks, such as <i>The Crown</i> and <i>Downton Abbey</i>.</p>
      <p>My spare time 2020-2021 was dedicated to learning web development.</p>
    </div>

  )
}