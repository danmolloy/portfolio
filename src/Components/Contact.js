import { useState } from "react"

export const Contact = () => {
  const [name, setName] = useState("")
  const [message, setMessage] = useState("")

  const sendMessage = () => {
    setName("");
    setMessage("");
  }

  return (
    <div id="contact" className="flex flex-col items-center contact flex flex-col p-12">
      <a href="mailto:danielmolloy_6@icloud.com" className="">Contact Me</a>
    </div>
  )
}