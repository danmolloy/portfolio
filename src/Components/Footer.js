
export const Footer = () => {

  const handleTop = () => {
    document.body.scrollTop = 0;
    document.documentElement.scrollTop = 0;
  }

  return (
    <div id="contact" className="bg-gray-100 flex flex-col justify-center items-center contact flex flex-col p-12">
      <a href="mailto:danielmolloy_6@icloud.com" className="no-underline hover:underline text-xl">Contact Me</a>
      <button onClick={() => handleTop()} className="no-underline hover:underline">Back to top</button>
    </div>
  )
}