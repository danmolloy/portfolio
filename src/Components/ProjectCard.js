export const ProjectCard = (props) => {
  return (
    <div className="project-card bg-white flex flex-col border m-8 w-auto items-center shadow">
      <h2 className="text-3xl mt-2">{props.project.title}</h2>
      <div className="flex flex-col md:flex-row items-center md:items-start md:justify-evenly m-4">
      <a href={props.project.link} target="_blank" rel="noreferrer" className="hover:shadow-md">
      <img src={props.project.image} alt="Project preview" className="w-80 border h-auto md:mr-2"/>
      </a>
      <div className="flex flex-col lg:flex-row">
      <div className="flex flex-col md:ml-2" >
        <p className="w-72">{props.project.blurb}</p>
        <div className="flex flex-row justify-between py-2">
          <a href={props.project.link}target="_blank" rel="noreferrer">Live Example</a>
          <a href={props.project.code} target="_blank" rel="noreferrer">Code</a>
        </div>
      </div>
      <div className="hidden md:flex flex-col ml-2 text-gray-400">
        <ul>
          {props.project.stack.map(i => 
            <li key={i}>{i}</li>)}
        </ul>
      </div>
      </div>
      </div>
    </div>
  )
}