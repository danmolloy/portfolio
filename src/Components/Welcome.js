export const Welcome = () => {
  return (
    <div className="welcome flex flex-col items-center justify-center">
      <img src="img/portrait.jpeg" className="max-h-screen-sm max-w-screen-sm md:self-end md:mr-20"/>
      <div className="absolute -mt-96 text-center text-white lg:text-black md:left-20">
        <h1 className="text-5xl md:text-6xl font-semibold z-10 text">Dan Molloy</h1>
        <h2 className="text-3xl md:text-4xl z-10 text">Front end developer</h2>
      </div>
    </div>
  )
}
