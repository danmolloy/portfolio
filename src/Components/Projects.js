import { ProjectCard } from "./ProjectCard"
import battleships from '../images/battleships.png'
import twitter from '../images/twitter.png'

const projects = [
   {
    title: "Twitter Clone",
    image: `${twitter}`,
    blurb: "A full stack social network, complete with authentication, authorization, notifications and direct messaging. Share your thoughts with your friends and interact with them.",
    link: "https://sharp-hugle-251ecd.netlify.app/",
    code: "https://github.com/danmolloy/twitter-clone",
    stack: ["React", "Apollo Client", "TypeScript", "Tailwind", "Jest", "React Router", "Node.js", "Apollo Server", "Prisma", "SQLIte",],
    key: 1
  },
  {
    title: "Battleships",
    image: `${battleships}`,
    blurb: "Play the classic board game against the computer. Go face-to-face with the computer to see who will win the battle. Try and beat the highscores!",
    link: "https://mystifying-saha-d28822.netlify.app/",
    code: "https://github.com/danmolloy/battleships",
    stack: ["React", "Redux", "Tailwind", "Typescript", "Jest", "Mirage JS"],
    key: 2
  }
]

export const Projects = () => {
  return (
    <div id="projects" className="flex flex-col items-center mt-24">
      <h1 className="text-4xl">Projects</h1>
      {projects.map(i => {
        return <ProjectCard project={i} key={i.key}/>
      })}
    </div>
  )
}