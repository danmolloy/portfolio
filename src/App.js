import './App.css';
import { About } from './Components/About';
import { Contact } from './Components/Contact';
import { Projects } from './Components/Projects';
import { Welcome } from "./Components/Welcome"

function App() {
  return (
    <div id="App" className="">
      <Welcome />
      <About /> 
      <Projects />
      <Contact />
    </div>
  );
}

export default App;
