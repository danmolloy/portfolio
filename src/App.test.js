import { act } from 'react-dom/test-utils';
import { unmountComponentAtNode, render } from 'react-dom';
import { Welcome} from "./Components/Welcome";
import { About } from "./Components/About";
import { Projects } from "./Components/Projects";
import { Footer } from "./Components/Footer";
import pretty from "pretty"
import App from './App';

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("Welcome renders without error", () => {
  act(() => {
    render(<Welcome />, container)
  })
  
  expect(container.textContent).toMatch(/Dan Molloy/gi)
  expect(pretty(container.innerHTML)).toMatchSnapshot();
})


it("About component renders", () => {
  act(() => {
    render(<About />, container)
  })
  expect(container.textContent).toMatch(/about/gi)
  expect(pretty(container.innerHTML)).toMatchSnapshot();
})

it("Projects component renders", () => {
  act(() => {
    render(<Projects />, container)
  })
  expect(container.textContent).toMatch(/projects/gi)
  expect(pretty(container.innerHTML)).toMatchSnapshot();
})

it("Footer component renders", () => {
  act(() => {
    render(<Footer />, container)
  })
  expect(container.textContent).toMatch(/contact/gi)
  expect(pretty(container.innerHTML)).toMatchSnapshot();
})