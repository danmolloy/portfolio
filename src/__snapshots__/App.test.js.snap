// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders 1`] = `
"<div id=\\"about\\" class=\\"text-black h-auto flex flex-col text-center justify-center my-24 mx-12\\">
  <h1 class=\\"p-4 text-4xl\\">About</h1>
  <div class=\\"flex flex-col itmes-center md:items-start md:flex-row justify-center \\"><img src=\\"bass.jpeg\\" alt=\\"Dan playing double bass\\" class=\\"h-auto w-full max-w-sm\\">
    <div class=\\"md:w-1/2 text-left md:ml-4\\">
      <div class=\\"p-2\\">
        <p>Originally from Australia, I moved to London in 2014.</p>
        <p>I have spent the past 12 years working as a classical double bassist, performing in some of the world's leading ensembles such as the London Symphony, BBC Symphony and Royal Philharmonic Orchestras.</p>
        <p>I've performed in every major concert hall in U.K. and I'm on a variety of soundtracks such as <i>The Crown</i>, <i>Downton Abbey</i> and <i>Seven Worlds, One Planet</i>.</p>
        <p>You can find out more about my music at <a href=\\"https://maslink.co.uk/client-directory?client=MOLLD1&amp;instrument=DOUBL1\\">Musician's Answering Service</a>.</p>
      </div>
      <div class=\\"p-2\\">
        <p>My spare time in 2020-2021 has been dedicated to learning web development. My focus has been on React and related libraries such as Redux, Apollo, Jest and TypeScript to name a few. </p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Footer component renders 1`] = `"<div id=\\"contact\\" class=\\"bg-gray-100 flex flex-col justify-center items-center contact flex flex-col p-12\\"><a href=\\"mailto:danielmolloy_6@icloud.com\\" class=\\"no-underline hover:underline text-xl\\">Contact Me</a><button class=\\"no-underline hover:underline\\">Back to top</button></div>"`;

exports[`Projects component renders 1`] = `
"<div id=\\"projects\\" class=\\"flex flex-col items-center mt-24\\">
  <h1 class=\\"text-4xl\\">Projects</h1>
  <div class=\\"project-card bg-white flex flex-col border m-8 w-auto items-center shadow\\">
    <h2 class=\\"text-3xl mt-2\\">Twitter Clone</h2>
    <div class=\\"flex flex-col md:flex-row items-center md:items-start md:justify-evenly m-4\\"><a href=\\"https://sharp-hugle-251ecd.netlify.app/\\" target=\\"_blank\\" rel=\\"noreferrer\\"><img src=\\"twitter.png\\" alt=\\"Project preview\\" class=\\"w-80 border h-auto md:mr-2\\"></a>
      <div class=\\"flex flex-col lg:flex-row\\">
        <div class=\\"flex flex-col md:ml-2\\">
          <p class=\\"w-72\\">A full stack social network, complete with authentication, authorization, notifications and direct messaging. Share your thoughts with your friends and interact with them.</p>
          <div class=\\"flex flex-col\\"><a href=\\"https://sharp-hugle-251ecd.netlify.app/\\" target=\\"_blank\\" rel=\\"noreferrer\\">Live Example</a><a href=\\"https://github.com/danmolloy/twitter-clone\\" target=\\"_blank\\" rel=\\"noreferrer\\">Code</a></div>
        </div>
        <div class=\\"hidden md:flex flex-col ml-2 text-gray-400\\">
          <ul>
            <li>React</li>
            <li>Apollo Client</li>
            <li>TypeScript</li>
            <li>Tailwind</li>
            <li>Jest</li>
            <li>React Router</li>
            <li>Node.js</li>
            <li>Apollo Server</li>
            <li>Prisma</li>
            <li>SQLIte</li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"project-card bg-white flex flex-col border m-8 w-auto items-center shadow\\">
    <h2 class=\\"text-3xl mt-2\\">Battleships</h2>
    <div class=\\"flex flex-col md:flex-row items-center md:items-start md:justify-evenly m-4\\"><a href=\\"https://mystifying-saha-d28822.netlify.app/\\" target=\\"_blank\\" rel=\\"noreferrer\\"><img src=\\"battleships.png\\" alt=\\"Project preview\\" class=\\"w-80 border h-auto md:mr-2\\"></a>
      <div class=\\"flex flex-col lg:flex-row\\">
        <div class=\\"flex flex-col md:ml-2\\">
          <p class=\\"w-72\\">Play the classic board game against the computer. Go face-to-face with the computer to see who will win the battle. Try and beat the highscores!</p>
          <div class=\\"flex flex-col\\"><a href=\\"https://mystifying-saha-d28822.netlify.app/\\" target=\\"_blank\\" rel=\\"noreferrer\\">Live Example</a><a href=\\"https://github.com/danmolloy/battleships\\" target=\\"_blank\\" rel=\\"noreferrer\\">Code</a></div>
        </div>
        <div class=\\"hidden md:flex flex-col ml-2 text-gray-400\\">
          <ul>
            <li>React</li>
            <li>Redux</li>
            <li>Tailwind</li>
            <li>Typescript</li>
            <li>Jest</li>
            <li>Mirage JS</li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`Welcome renders without error 1`] = `
"<div class=\\"welcome flex flex-col items-center justify-center\\"><img src=\\"portrait.jpeg\\" alt=\\"Dan in a garden\\" class=\\"max-h-screen-sm max-w-screen-sm md:self-end md:mr-20\\">
  <div class=\\"absolute sm:-mt-96 text-center text-white lg:text-black md:left-20\\">
    <h1 class=\\"text-5xl md:text-6xl font-semibold z-10 text\\">Dan Molloy</h1>
    <h2 class=\\"text-3xl md:text-4xl z-10 text\\">Front end developer</h2>
  </div>
</div>"
`;
